build_cmake/CMakeLists.txt

```diff
@@ -24,7 +24,7 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 if (NOT TARGET Framework)
-	add_subdirectory(../../Framework/build_cmake ${CodeGen_SOURCE_DIR}/../../Framework/build_cmake/build)
+	add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake ${CMAKE_CURRENT_SOURCE_DIR}/../../Framework/build_cmake/build)
```

Comment:
was rushing to leave the house yesterday, completely missed that :/ so ill have a look at this now, again, to make sure it works this time :p

as for the arch type, cmake doesn't have a native way of detecting target arch, and it differs between build, e.g on windows ${ARCH}, on android ${CMAKE_ANDROID_ARCH_ABI}, on linux none that i can find and on OSX ${CMAKE_OSX_ARCHITECTURES} but only if set by the user